docs/source/json-api/index.rst

```diff
@@ -32,140 +32,123 @@ We welcome feedback about the JSON API on `our issue tracker
-subprotocol.
+Authentication
+--------------
 
-In testing environments, you can use https://jwt.io to generate your
+The authentication token is used exclusively for maintaining the internal list of known packages and templates.
+
+.. note:: At no point should an authentication token be provided to an end user, these are for internal use only.
+
+Every authentication token is different and will depend on what your specific ledger or ledger operator requires.
+The JSON API server requires no access to party-specific data, only access to the ledger identity and package services.
+A token issued for the HTTP JSON API server should contain enough claims to contact these two services but no more than that.
+Please refer to your ledger operator's documentation to find out how.
+
+Once you have retrieved your access token, you can provide it to the JSON API by storing it in a file
+and starting ``daml json-api`` with the flag ``--access-token-file /path/to/your/token.file``.
+
+If the token cannot be read from the provided path or the Ledger API reports an authentication error
+(for example due to token expiration), the JSON API will report the error via logging. 
+
+.. note:: If the token file is updated with a new token it will be picked up at the next attempt to send a request. You can use this to handle cases where an old token expires without interrupting your JSON API service.
+
+Authorization
+-------------
+
+Every request from a client to the JSON API requires you to specify a party and some other settings,
+with a JWT token.  Normal HTTP requests pass the token in an ``Authorization`` header, 
+while WebSocket requests pass the token in a subprotocol.
+
+.. note:: While the JSON API receives the token it doesn't handle it itself. Upon receiving a token it will pass it on to the underlying Ledger's AuthService which will then determine if the token is valid and authorized. As such each token format is ledger/implementation-specific and the below example reflects only the format that is used for the DAML Sandbox.
+
+In the DAML Sandbox testing environment, you can use https://jwt.io to generate your
 token.  The default "header" is fine.  Under "Payload", fill in:
 
 .. code-block:: json
 
     {
       "https://daml.com/ledger-api": {
         "ledgerId": "MyLedger",
-        "applicationId": "foobar",
+        "applicationId": "HTTP-JSON-API-Gateway",
         "actAs": ["Alice"]
       }
     }
 
-Keep in mind that the value of the ``ledgerId`` payload field has to match the one passed to the sandbox with the ``--ledgerid`` argument. You can replace ``Alice`` with whatever party you want to use.
+The value of the ``ledgerId`` field has to match the ``ledgerId` of your underlying DAML Ledger.
+For the ``daml sandbox`` this corresponds to the ``--ledgerid MyLedger`` flag.
+
+The value of the ``applicationId`` field must match that provided to the ``daml json-api`` command with the flag
```

Comment:
I don’t think this is true. As far as I can tell the `applicationId` you pass in via the command line flag is just unused atm. It definitely doesn’t need to match the one in the token (I wrote a test for that last week).

Reply:
If `applicationId` doesn't need to match anything that's pretty confusing for end users.

I believe `ledgerId` needs to match the DAML Ledger though.